### PR TITLE
Update peer state and sc to protocol

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Update peers state and sc to protocol in sovereign chain block processing
  
## Proposed changes
- Implemented `updateShardDataPeerState` to work **only for current header.** Original Implementation (for metachain) had metaBlock notarized shard headers from cache to be used when updating validators data (current block -1).
- `validatorStatisticsProcessor.UpdatePeerState` in sov chain processing at `processEpochStartMetaBlock` level
- `scToProtocol.UpdateProtocol` called for `postProcessMBs` after processing all sovereign txs

## Testing procedure
- Integration tests with staking new nodes which also go through auction

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: cmd/sovereignnode/chainSimulator/tests/epochChange/epochChange_test.go [hunk 1]
```diff
@@ -55,13 +64,35 @@ func TestSovereignChainSimulator_EpochChange(t *testing.T) {
 
 	defer cs.Close()
 
-	trie := cs.GetNodeHandler(0).GetStateComponents().TriesContainer().Get([]byte(dataRetriever.PeerAccountsUnit.String()))
+	nodeHandler := cs.GetNodeHandler(0)
```
Comment (axenteoctavian) [inline cmd/sovereignnode/chainSimulator/tests/epochChange/epochChange_test.go:67]: Would be better to use `core.SovereignChainShardId` instead of `0`

File: factory/processing/processComponents_test.go [hunk 2]
```diff
@@ -1012,6 +1023,8 @@ func getRunTypeComponents(rt runType.RunTypeComponentsHolder) *mainFactoryMocks.
 		GenesisMetaBlockChecker:             rt.GenesisMetaBlockCheckerCreator(),
 		NodesSetupCheckerFactoryField:       rt.NodesSetupCheckerFactory(),
 		EpochStartTriggerFactoryField:       rt.EpochStartTriggerFactory(),
+		LatestDataProviderFactoryField:      rt.LatestDataProviderFactory(),
```
Comment (axenteoctavian) [inline factory/processing/processComponents_test.go:1026]: :+1: 
Can we also have a test for nil `LatestDataProviderFactoryField`? We missed this in the previous PR

File: cmd/sovereignnode/chainSimulator/tests/epochChange/epochChange_test.go (hunk 1 above)
Comment (mariusmihaic) [reply cmd/sovereignnode/chainSimulator/tests/epochChange/epochChange_test.go:67]: Right, did as suggested

File: factory/processing/processComponents_test.go (hunk 2 above)
Comment (mariusmihaic) [reply factory/processing/processComponents_test.go:1026]: Test was added in previous PR

**Review by axenteoctavian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]
